File: 프로그래머스/3/43163. 단어 변환/단어 변환.js
function solution(begin, target, words) {
    let answer = 0;
    let visited = Array(words.length).fill(false)
    
    if(!words.includes(target)){
        return 0;
    }
    
    let queue = [];
    queue.push([begin,0]);

    while(queue){
        let [word,cnt] = queue.shift()
        
        if(word === target){
            answer = cnt
            break
        } else {
            for(var i=0; i<words.length; i++){
                let tmp_cnt = 0
                if(!visited[i]){
                    for(var j=0; j<words.length; j++){
                        if(word[j] !== words[i][j]) {
                            tmp_cnt += 1
                        }
                    }
                }
                if(tmp_cnt === 1){
                    queue.push([words[i],cnt+1])
                    visited[i] = true
                }
            }
        }
    }
    return answer;
}

File: 프로그래머스/3/43163. 단어 변환/단어 변환.py
from collections import deque
def solution(begin, target, words):
    answer = 0
    visited = [0 for _ in range(len(words))]
    if target not in words:
        answer = 0
        
    q = deque()
    q.append([begin,0]) #단어, 변환 수 
    while q:
        word,cnt = q.popleft()
        if word == target:
            answer = cnt
            break
        for i in range(len(words)):
            tmp_cnt = 0
            if visited[i] == 0:
                for j in range(len(word)):
                    if word[j] != words[i][j]:
                        tmp_cnt += 1
            if tmp_cnt == 1:
                q.append([words[i],cnt+1])
                visited[i] = 1
        
    return answer